### PR TITLE
Add opencensus propagation readme and example

### PR DESCRIPTION
**Pull request by dashpole**
Issue: open-telemetry/opentelemetry-go#93

Demonstrate propagation from an opencensus client to an opentelemetry server.

### CODE REVIEW

File: propagators/opencensus/examples/go.mod [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module go.opentelemetry.io/opentelemetry-go/bridge/opencensus/examples/grpc
```
Comment (MrAlias) [inline propagators/opencensus/examples/go.mod:1]:
```suggestion
module go.opentelemetry.io/contrib/propagators/opencensus/examples
```

File: propagators/opencensus/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+## The Problem
+
+The [ocgrpc](https://github.com/census-instrumentation/opencensus-go/tree/master/plugin/ocgrpc) GRPC plugin for OpenCensus is hard-coded to use a [Binary propagation format](https://github.com/census-instrumentation/opencensus-go/blob/380f4078db9f3ee20e26a08105ceecccddf872b8/trace/propagation/propagation.go).
+
+A GRPC client and server that use OpenCensus cannot easily migrate to OpenTelemetry because there will be a period of time during which one will use OpenCensus and the other will use OpenTelemetry.  If both client and server export spans to the same trace backend, the server spans will not be a child of the client spans, because they are using different propagation formats.  To be able to easily migrate from OpenCensus to OpenTelemetry, it is necessary to use the OpenCensus binary propagation format with OpenTelemetry.
+
+## Usage
+
+To add use the binary propagation format with otelgrpc, use the WithPropagators option to the otelgrpc Interceptors:
```
Comment (MrAlias) [inline propagators/opencensus/README.md:11]:
```suggestion
To use the binary propagation format with otelgrpc, use the WithPropagators option to the otelgrpc Interceptors:
```

File: propagators/opencensus/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+Before:
```
Comment (MrAlias) [inline propagators/opencensus/README.md:13]: Is this a leftover of a previous edit where there was a before and after?

File: propagators/opencensus/examples/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# OpenCensus binary propagation example
+
+The server uses OpenTelemetry with the opencensus binary propagation format.
```
Comment (MrAlias) [inline propagators/opencensus/examples/README.md:3]:
```suggestion
The server uses OpenTelemetry with the OpenCensus binary propagation format.
```

right?

File: propagators/opencensus/examples/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+The client uses OpenCensus, which is hard-coded to use the opencensus binary
```
Comment (MrAlias) [inline propagators/opencensus/examples/README.md:4]:
```suggestion
The client uses OpenCensus, which is hard-coded to use the OpenCensus binary
```

File: propagators/opencensus/examples/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+First, start the opentelemetry server:
+```bash
+go run opentelemetry_server/server.go
+```
+
+In another shell, start the OpenCensus client:
+```bash
+go run opencensus_client/client.go
+```
```
Comment (MrAlias) [inline propagators/opencensus/examples/README.md:19]: Can we add a sentence or two about what to expect from the example? Like expected STDOUT output?

File: propagators/opencensus/examples/opencensus_client/client.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+	"google.golang.org/grpc"
+)
+
+const (
+	address     = "localhost:50051"
+	defaultName = "world"
+)
+
+func main() {
+	log.Println("Configuring opencensus, and registering the Print exporter.")
```
Comment (MrAlias) [inline propagators/opencensus/examples/opencensus_client/client.go:36]:
```suggestion
	log.Println("Configuring OpenCensus, and registering the Print exporter.")
```

File: propagators/opencensus/examples/opentelemetry_server/server.go [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+	return &pb.HelloReply{Message: "Hello " + in.Name}, nil
+}
+
+func main() {
+	lis, err := net.Listen("tcp", port)
+	if err != nil {
+		log.Fatalf("Failed to listen: %v", err)
+	}
+
+	log.Println("Registering opentelemetry stdout exporter.")
```
Comment (MrAlias) [inline propagators/opencensus/examples/opentelemetry_server/server.go:54]:
```suggestion
	log.Println("Registering OpenTelemetry stdout exporter.")
```

**Review by MrAlias — CHANGES_REQUESTED** [review]
Only blocking issue is the go.mod package name.

Overall looks good though.

File: propagators/opencensus/README.md (hunk 3 above)
Comment (dashpole) [reply propagators/opencensus/README.md]: Yes

File: propagators/opencensus/examples/README.md (hunk 6 above)
Comment (dashpole) [reply propagators/opencensus/examples/README.md:19]: Done.

**Review by XSAM — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**